vault/data_source_kv_secret.go

```diff
@@ -70,7 +76,8 @@ func kvSecretDataSourceRead(ctx context.Context, d *schema.ResourceData, meta in
 		return diag.Errorf("no secret found at %q", path)
 	}
 
-	jsonData, err := json.Marshal(secret.Data)
+	data := secret.Data
```

Comment:
nit: think you can just inline `secret.Data` rather than adding the extra level of indirection here.